api/environments/job_handler.go

```diff
@@ -239,6 +241,36 @@ func (eh EnvironmentHandler) DeleteBatch(ctx context.Context, appName, envName,
 	return eh.accounts.UserAccount.RadixClient.RadixV1().RadixBatches(batch.GetNamespace()).Delete(ctx, batch.GetName(), metav1.DeleteOptions{})
 }
 
+// CopyBatch Copy batch by name
+func (eh EnvironmentHandler) CopyBatch(ctx context.Context, appName, envName, jobComponentName, batchName string, scheduledBatchRequest environmentModels.ScheduledBatchRequest) (*deploymentModels.ScheduledBatchSummary, error) {
+	deploymentName := scheduledBatchRequest.DeploymentName
+	jobSchedulerBatchHandler := eh.jobSchedulerHandlerFactory.CreateJobSchedulerBatchHandlerForEnv(getJobSchedulerEnvFor(appName, envName, jobComponentName, deploymentName))
+	batchStatus, err := jobSchedulerBatchHandler.CopyBatch(ctx, batchName, deploymentName)
+	if err != nil {
+		return nil, err
+	}
+	return eh.getScheduledBatchStatus(batchStatus, deploymentName), nil
+}
+
+// CopyJob Copy job by name
+func (eh EnvironmentHandler) CopyJob(ctx context.Context, appName, envName, jobComponentName, jobName string, scheduledJobRequest environmentModels.ScheduledJobRequest) (*deploymentModels.ScheduledJobSummary, error) {
+	deploymentName := scheduledJobRequest.DeploymentName
+	jobSchedulerJobHandler := eh.jobSchedulerHandlerFactory.CreateJobSchedulerJobHandlerForEnv(getJobSchedulerEnvFor(appName, envName, jobComponentName, deploymentName))
+	jobStatus, err := jobSchedulerJobHandler.CopyJob(ctx, jobName, deploymentName)
+	if err != nil {
+		return nil, err
+	}
+	batchStatus := deploymentModels.ScheduledJobSummary{
+		Name:           fmt.Sprintf("%s-%s", jobStatus.BatchName, jobStatus.Name),
+		DeploymentName: deploymentName,
+		BatchName:      jobStatus.BatchName,
+		JobId:          jobStatus.JobId,
+		Status:         jobStatus.Status,
+	}
+
+	return &batchStatus, nil
+}
```

Comment:
CopyBatch builds the response using a method, whereas CopyJob builds it inline.
Would be more consistent to choose one of them. I suggest inlining it since the method doesn't do much, and it is only used by CopyBatch

Reply:
Corrected with extracted method